Imageview/app/src/main/java/com/mycompany/imageview/MySQLiteHelper.java

```diff
@@ -0,0 +1,180 @@
+package com.mycompany.imageview;
+
+/**
+ * Created by root on 9/28/15.
+ */
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+import android.util.Log;
+
+import java.io.File;
+import java.util.LinkedList;
+import java.util.List;
+
+public class MySQLiteHelper extends SQLiteOpenHelper {
+
+    // Database Version
+    private static final int DATABASE_VERSION = 2;
+    // Database Name
+    private static final String DATABASE_NAME = "Image_Library";
+
+    // Contacts table name
+    private static final String TABLE_CONTACTS = "Images";
+    // Image Table Columns names
+    private static final String KEY_ID = "id";
+    private static final String KEY_keyword = "keyword";
+    private static final String KEY_link = "link";
+
+    private static final String[] COLUMNS = {KEY_ID,KEY_keyword,KEY_link};
+
+
+
+
+
+
+    public MySQLiteHelper(Context context) {
+        super(context, DATABASE_NAME, null, DATABASE_VERSION);
+    }
+
+    @Override
+    public void onCreate(SQLiteDatabase db) {
+        // SQL statement to create book table
+        String CREATE_IMAGE_TABLE = "CREATE TABLE "+TABLE_CONTACTS+" ( " +
+                KEY_ID+" INTEGER PRIMARY KEY AUTOINCREMENT, " +KEY_keyword+
+                " TEXT, "+
+                KEY_link+" TEXT )";
+
+        // create books table
+        db.execSQL(CREATE_IMAGE_TABLE);
+    }
+
+    @Override
+    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+        // Drop older books table if existed
+        db.execSQL("DROP TABLE IF EXISTS "+ TABLE_CONTACTS);
+
+        // create fresh books table
+        this.onCreate(db);
+    }
+
+
+
+
+    public void add_image(Image_DB image){
+        //for logging
+        Log.d("add_image", image.toString());
+
+        // 1. get reference to writable DB
+        SQLiteDatabase db = this.getWritableDatabase();
```

Comment:
You will get a a performance decrease if you have to get the writable database each time you add an image. If you create a Database object as a class parameter and instantiate it in the oncreate, you will have less calls to getWriteableDatabase().